runtime-sdk/modules/evm/src/raw_tx.rs

```diff
@@ -178,7 +178,8 @@ pub fn decode(
             fee: transaction::Fee {
                 amount: token::BaseUnits(resolved_fee_amount, token::Denomination::NATIVE),
                 gas: gas_limit,
-                consensus_messages: 0,
+                // TODO: Allow customization, maybe through call data?
+                consensus_messages: 1,
```

Comment:
If no messages are actually emitted it doesn't do anything. The only problem is that in case the per-block consensus message quota is used up, such transaction would not be scheduled even if they end up not emitting any messages. However I don't see a nice way to handle this currently unless we add an extension to regular calls to annotate them.

Reply:
does the execute-while-scheduling system waste consensus messages if transactions have this limit 1 set but don't use it?

Reply:
It does not, only actually emitted messages count towards the quota. So the above problem is only relevant if the quota is full because of actually emitted messages -- then even transactions that don't emit messages but have this field set in the fee structure would not be scheduled.

Reply:
huh, ok